pubsub/api/Pubsub.Samples/ExactlyOnceDeliverySubscriberAsync.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright 2022 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START pubsub_subscriber_exactly_once]
+
+using Google.Cloud.PubSub.V1;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Google.Cloud.PubSub.V1.SubscriberClient;
+
+public class ExactlyOnceDeliverySubscriberAsyncSample
+{
+    public async Task<int> ExactlyOnceDeliverySubscriberAsync(string projectId, string subscriptionId)
+    {
+        SubscriptionName subscriptionName = SubscriptionName.FromProjectSubscription(projectId, subscriptionId);
+        SubscriberClient subscriber = await SubscriberClient.CreateAsync(subscriptionName);
+        // To get the status of ACKnowledge (ACK) or Not ACKnowledge (NACK) request in exactly once delivery subscriptions,
+        // create a new handler that implements the SubscriptionHandler. The handler is backward compatible
+        // and can be used with non-exactly once delivery subscriptions as well.
+        var subscriptionHandler = new SampleSubscriptionHandler();
+        Task startTask = subscriber.StartAsync(subscriptionHandler);
+        // Run for 5 seconds.
+        await Task.Delay(5000);
+        await subscriber.StopAsync(CancellationToken.None);
+        // Let's make sure that the start task finished successfully after the call to stop.
+        // subscriptionHandler's HandleAckResponses will have the response for every ACKnowledge request that is successful or failed.
```

Comment:
Hey Amanda, The method HandleAckResponses has a parameter of type IReadOnlyList<AckNackResponse> which will contain the successful or permanently failed AckIds. There may be a few messages with temporary errors as well, but they won't appear in the list unless they succeed or fail permanently. If have isn't appropriate, please suggest a suitable comment to convey this.

Reply:
Yes, OK, but still the method won't "have" anything, possibly "receive" instead?
And also, it is not clear why this comment is in this place? Will the method be called only after the susbscriber has stopped? Can it be called several times while the subscriber is listening, etc. but making sure that the start task is finished guarantees that it will be called for a last time?

Reply:
Sure. Will update it with "receive". 
Right Amanda. The handler can be called several times while the subscriber is listening. Essentially, everytime we get a response from the Ack/Nack request this handler will be called.

Reply:
Ok, so maybe move this comment then to the handler itself? The code of it will part of the sample so that's fine.

Reply:
Sure.